### PR TITLE
Script: ensure prevrandao is set, even if no mixHash in response

### PR DESCRIPTION
**Pull request by 0xCalibur**
## Motivation
ref https://github.com/foundry-rs/foundry/issues/4238

ensure prevrandao is set, even if no mixHash in response

## Solution
same fix as anvil but when running forge script.

### CODE REVIEW

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Review by mattsse — CHANGES_REQUESTED** [review]
this doesn't compile, 

please put this in a `Some`

please consider enabling changes by maintainer

**Comment by 0xm3rlin** [comment]
> this doesn't compile,
> 
> please put this in a `Some`
> 
> please consider enabling changes by maintainer

the opposite might be helpful as the changes by @DaniPopes @Evalir were errorful.

i would suggest reverting this pr back to the original commit 50212024e134b4117b17145e3999bf14e775418a by @0xCalibur

**Comment by Evalir** [comment]
hey @0xm3rlin — yeah we committed a suggestion which was errorful. That being said, we can't push to this branch as the PR comes from the fork's master, which is protected. Happy to merge this ASAP if @0xCalibur fixes it—it's just a one line change. If not happy to close and reopen on another PR to merge this quickly.

**Comment by 0xCalibur** [comment]
> hey @0xm3rlin — yeah we committed a suggestion which was errorful. That being said, we can't push to this branch as the PR comes from the fork's master, which is protected. Happy to merge this ASAP if @0xCalibur fixes it—it's just a one line change.

I'm confused...so I just need to go back to the original changes?

**Review by Evalir — CHANGES_REQUESTED** [review]

File: evm/src/executor/fork/init.rs
Comment (mattsse) [inline evm/src/executor/fork/init.rs:70]:
```suggestion
            prevrandao: Some(block.mix_hash.map(u256_to_ru256).unwrap_or_default()),
```

**Review by mattsse — APPROVED** [review]
okay, after several rounds of reviews for this one liner, I think we finally figured it out -.-

**Comment by mattsse** [comment]
well, this has been more effort than expected lol

I'm closing this, and pushed changes manually to #4929

can't make changes manually here since this is opened from master branch

**Comment by DaniPopes** [comment]
oops

**Pull request closed without merging**
